src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,258 @@
+import { useEffect, useState, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { ErrorOption, Person, SortBy } from '../types';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { SearchLink } from './SearchLink';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorOption, setErrorOption] = useState(ErrorOption.NoError);
+  const PEOPLE_URL
+  = 'https://mate-academy.github.io/react_people-table/api/people.json';
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const resetAll = () => {
+    return {
+      centuries: null,
+      query: null,
+      sex: null,
+      sort: null,
+      order: null,
+    };
+  };
+
+  const getCentury = (bornYear: number) => Math.ceil(bornYear / 100);
+
+  const generateClass = (sortBy: string) => {
+    return cn('fas fa-sort', {
+      'fa-sort-up': sort === sortBy && !order,
+      'fa-sort-down': sort === sortBy && order,
+    });
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+
+    if (event.target.value) {
+      params.set('query', event.target.value);
+    } else {
+      params.delete('query');
+    }
+
+    setSearchParams(params.toString());
+  };
+
+  const handleCenturiesChange = (century: string) => {
+    return centuries.includes(century)
+      ? centuries.filter((c) => c !== century)
+      : [...centuries, century];
+  };
+
+  const handleSortChange = (sortBy: string) => {
+    let result;
+    const isSortBy = sort === sortBy;
+    const isDesc = order === 'desc';
+
+    if (isSortBy && isDesc) {
+      result = { sort: null, order: null };
+    } else if (!isSortBy && !isDesc) {
+      result = { sort: sortBy, order: null };
+    } else if (!isSortBy && isDesc) {
```

Comment:
```suggestion
    if (isSortBy && isDesc) {
      result = { sort: null, order: null };
      
      return result;
    } 
    
    if (!isSortBy && !isDesc) {
      result = { sort: sortBy, order: null };
      
      return result;
    }
```

Reply:
your code isn't working

Reply:
@Reject-4444  you can put value in `else` block to variable declaration(row 62)